components/places/src/api/places_api.rs

```diff
@@ -264,15 +262,36 @@ impl PlacesApi {
 
         result?;
 
-        Ok(sync_ping)
+        Ok(())
+    }
+
+    pub fn sync_history(
```

Comment:
why do we need both `sync_history` and `sync_history_with_telemetry`?

Reply:
I could inline `sync_history_with_telemetry` into the closure I pass to `SyncTelemetryPing::record`.